### PR TITLE
CLN/TST: parametrize some tests in tests.indexing.test_float

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/indexing/test_floats.py [hunk 1]
```diff
@@ -32,19 +32,7 @@ def check(self, result, original, indexer, getitem):
...
-            tm.makeUnicodeIndex,
-            tm.makeCategoricalIndex,
-            tm.makeDateIndex,
-            tm.makeTimedeltaIndex,
-            tm.makePeriodIndex,
-            tm.makeIntIndex,
-            tm.makeRangeIndex,
-        ],
-    )
+    @pytest.mark.parametrize("index_func", tm.all_index_generator(5))
```
Comment (jreback) [inline pandas/tests/indexing/test_floats.py]: can you use the indices fixture?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_floats.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/indexing/test_floats.py]: not without being pretty invasive

Comment (jreback) [reply pandas/tests/indexing/test_floats.py]: kk, on the list for the future

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm merge on green.

**Pull request merged**